python/cudf/cudf/core/dataframe.py

```diff
@@ -5480,7 +5482,16 @@ def from_arrow(cls, table):
                     # https://github.com/apache/arrow/issues/15178
                     out = out.set_index(idx)
             else:
-                out = out.set_index(index_col[0])
+                out = out.set_index(index_col)
+
+        if "__index_level_0__" in out.index.names:
+            assert len(out.index.names) == 1
```

Comment:
```suggestion
        if "__index_level_0__" in out.index.names and len(out.index.names) == 1:
```

Should we not raise and rather do this instead?

Reply:
Yeah, I think that would be fine. I suppose it is *possible* for one column in a multi-index to be called `"__index_level_0__"` :)